src/providers/ldap/sdap_async_nested_groups.c

```diff
@@ -1412,20 +1414,54 @@ sdap_nested_group_single_send(TALLOC_CTX *mem_ctx,
     return req;
 }
 
+static bool must_ignore(struct sdap_nested_group_single_state *state)
+{
+    bool ignore;
+    const char *dn;
+    const char *postfix;
+    struct sdap_search_base **base;
+    int diff;
+
+    DEBUG(SSSDBG_TRACE_INTERNAL, "[TEST] Starting with DN %s\n", state->current_member->dn);
+
+    dn = state->current_member->dn;
+    ignore = false;
+    DEBUG(SSSDBG_TRACE_INTERNAL, "[TEST] ignore_search_bases is %p\n", state->group_ctx->ignore_search_bases);
+    if (state->group_ctx->ignore_search_bases != NULL) {
+        for (base = state->group_ctx->ignore_search_bases; *base != NULL; base++) {
+            postfix = (*base)->basedn;
+            DEBUG(SSSDBG_TRACE_INTERNAL, "[TEST] Trying postfix %s\n", postfix);
+            diff = strlen(dn) - strlen(postfix);
+            if (diff >= 0) {
+                ignore = (strcmp(postfix, dn + diff) == 0);
```

Comment:
This should at least use `strcasecmp()` since the attribute names in LDAP DNs are case-insensitive, e.g. 'DC=ad,DC=dom', 'dc=ad,dc=dom' or 'DC=ad,dc=dom' are all the same. If the attribute values are case-insentive or not depends on the schema definition of the attribute type but here I think we can safely assume that the case-insensitive comparison will cause no issues.

However, it might be even better to use dedicated functions for DN comparison. `libdlb` provides `ldb_dn_compare_base()` to check if a given DN is inside a given base DN. The DN strings have to be converted into `struct ldb_dn`. So to speed things up it would make sense to translate the ignore DNs into `struct ldb_dn` during the initialization, the strings can be kept as well for debug messages.

Reply:
Now using `ldb_dn_compare_base()`. And thanks for the tip about `sysdb_ctx_get_ldb()`.